src/parse.ts

```diff
@@ -112,7 +112,12 @@ function parseAgainstNode(reader: StringReader, node: CommandNode,
     const parser = getParser(node);
     const helper = new ReturnHelper();
     if (!!parser) {
-        const result = parser.parse(reader, createParserInfo(node, data, path, context, false));
+        const contextData = !parser.contextPaths ? undefined :
```

Comment:
As I previously said in private discussions, this method has little to no point. If the parser wants context. However, I wouldn't mind you splitting out the path equality testing